ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/errors/KVErrors.scala

```diff
@@ -342,122 +259,4 @@ object KVErrors extends LedgerApiErrorGroup {
     }
 
   }
-
-  @Explanation(
-    "A command for the same ledger change has already been successfully processed " +
-      "within the current deduplication window."
-  )
-  @Resolution("Celebrate, as your command has already been processed.")
-  object DuplicateCommand
```

Comment:
I am not convinced `CommandPreparation.DuplicateCommand` is the same thing semantically as committer-discovered duplicate. I would suggest to create another error `CommandRejections.DuplicateCommand`

Reply:
Would you have a different error code id for DuplicateCommand or just a different case class in the same error code?

Reply:
I agree `CommandRejections.DuplicateCommand` would be a good candidate to move the committer-deduplicated rejection. From a client's perspective I'm not sure if there's a practical difference (i.e., there's no point in retrying the command as it will most probably rejected again), however, committer vs. participant deduplicated commands should be distinguishable at least via error IDs as they are generated at different parts of the workflow and have different guarantees.
@andreaslochbihler-da Any thoughts on this?

Reply:
I agree with @mziolekda that the prefix `CommandPreparation` is misleading here. Duplicated commands really are command submission errors. From an application point of view, however, they should be treated in the same way though, and therefore there's no point in assigning different error codes to them. However, every ledger implementation may want to provide different metadata. For that, different `Reject` case classes can be used. For example, Canton creates its own case class in https://github.com/DACH-NY/canton/blob/master/community/participant/src/main/scala/com/digitalasset/canton/participant/sync/CommandDeduplicationError.scala#L35.

Reply:
> however, committer vs. participant deduplicated commands should be distinguishable at least via error IDs as they are generated at different parts of the workflow and have different guarantees.

@miklos-da In what sense do they provide different guarantees?

Reply:
cd3e60acba627cc69f640218a1914d9e66cba86e Moved `DuplicatedCommand` to `CommandRejections`